### DISCUSSION TITLE
TypeError: 'Session' object is not callable when trying to post.

### DISCUSSION THREAD

**Discussion by stigrunar** [fastapi/fastapi#6773] [Questions] [answered]
Trying to follow the tutorial in the fastapi documentation and implementing it into my own testproject.

--------------------------------------------------------------------

Code:
```Python
from streamticket import app, templates
from fastapi import Request, Depends
from ..models import Keys
from pydantic import BaseModel
from sqlalchemy.orm import Session
from streamticket.database import SessionLocal, engine


class KeyRequest(BaseModel):
   customer: str
   amount: int

def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

@app.get ("/")
async def index(request: Request):
    return templates.TemplateResponse("index.html", {"request": request})

@app.get ("/login")
async def about(request: Request):
    return templates.TemplateResponse("login.html", {"request": request})

@app.post("/keygenerator")
def create_key(key_request: KeyRequest, db: Session = Depends(get_db)):

    keys = Keys()
    keys.customer = key_request.customer
    keys.amount = key_request.amount

    db.add(keys)
    db.commit()

    return {
        "code": "success",
        "message": "key created"
    }
```

### Description

* Using insomnia  to create a post on `/keygenerator`.
* Expecting. {
        "code": "success",
        "message": "key created"
    }
* Error:
" INFO:     127.0.0.1:53349 - "POST /keygenerator HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
TypeError: 'Session' object is not callable when trying to post.

### Environment

* OS: [ Windows ]:
* FastAPI Version latest
* Python version 3.9

**Comment by ghost** [comment, marked as answer]
I was expecting a `SessionLocal` instance created with `sessionmaker` as folllow;

```Python
from sqlalchemy import create_engine
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

SQLALCHEMY_DATABASE_URL = "sqlite:///./sql_app.db"
# SQLALCHEMY_DATABASE_URL = "postgresql://user:password@postgresserver/db"

engine = create_engine(
    SQLALCHEMY_DATABASE_URL, connect_args={"check_same_thread": False}
)
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)

Base = declarative_base()

```

You may want to check `https://fastapi.tiangolo.com/tutorial/sql-databases/`

**Comment by stigrunar** [comment]
> I was expecting a `SessionLocal` instance created with `sessionmaker` as folllow;
> 
> ```python
> from sqlalchemy import create_engine
> from sqlalchemy.ext.declarative import declarative_base
> from sqlalchemy.orm import sessionmaker
> 
> SQLALCHEMY_DATABASE_URL = "sqlite:///./sql_app.db"
> # SQLALCHEMY_DATABASE_URL = "postgresql://user:password@postgresserver/db"
> 
> engine = create_engine(
>     SQLALCHEMY_DATABASE_URL, connect_args={"check_same_thread": False}
> )
> SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
> 
> Base = declarative_base()
> ```
> 
> You may want to check `https://fastapi.tiangolo.com/tutorial/sql-databases/`

Yes i have the sessionlocal instance in my database.py.

**Comment by tiangolo** [comment]
I assume you solved it, thanks for closing the issue 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
